pseudomonas_analysis/nbconverted/viz_expression_data.py

```diff
@@ -0,0 +1,579 @@
+# 2. To visualize the signal that is in the experiments
+
+# In[1]:
+
+
+get_ipython().run_line_magic('load_ext', 'autoreload')
+get_ipython().run_line_magic('load_ext', 'rpy2.ipython')
+get_ipython().run_line_magic('autoreload', '2')
+
+
+# In[2]:
+
+
+import os
+import pandas as pd
+import umap
+import pickle
+import glob
+import seaborn as sns
+from sklearn.decomposition import PCA
+from keras.models import load_model
+import plotnine as pn
+
+from ponyo import utils
+from generic_expression_patterns_modules import plot
+
+
+# ## Load config parameters
+
+# In[3]:
+
+
+# Read in config variables
+base_dir = os.path.abspath(os.path.join(os.getcwd(), "../"))
+
+config_filename = os.path.abspath(
+    os.path.join(base_dir, "configs", "config_pseudomonas_33245.tsv")
+)
+
+params = utils.read_config(config_filename)
+
+
+# In[4]:
+
+
+# Load config params
+
+local_dir = params['local_dir']
+project_id = params['project_id']
+num_simulated = params['num_simulated']
+
+pval_name = "adj.P.Val"
+logFC_name = "logFC"
+run=0
+
+# Settings for running visualization using pseudomonas config file
+vae_model_dir = os.path.join(base_dir,"pseudomonas_analysis", "models", "NN_2500_30")
+template_filename = params['processed_template_filename']
+normalized_compendium_filename = params['normalized_compendium_filename']
+scaler_filename = params['scaler_filename']
+
+
+# ## Volcano plots
+
+# In[5]:
+
+
+# Check number of DEGs
+template_DE_stats_filename = os.path.join(
+    local_dir,
+    "DE_stats",
+    f"DE_stats_template_data_{project_id}_real.txt"
+)
+
+template_DE_stats = pd.read_csv(
+    template_DE_stats_filename, 
+    sep="\t", 
+    header=0, 
+    index_col=0
+)
+
+selected = template_DE_stats[(template_DE_stats[pval_name]<0.01) & (abs(template_DE_stats[logFC_name])>1)]
+print(selected.shape)
+
+
+# In[6]:
+
+
+plot.make_volcano_plot_template(
+    template_DE_stats_filename,
+    project_id,
+    pval_name,
+    logFC_name
+)
+
+
+# In[7]:
+
+
+simulated_DE_stats_dir = os.path.join(local_dir, "DE_stats")
+
+plot.make_volcano_plot_simulated(
+    simulated_DE_stats_dir,
+    project_id,
+    pval_name,
+    logFC_name,
+    num_simulated,
+    5,
+    5,
+    20,
+    15
+)
+
+
+# ## Explore flat volcano plots
+# 
+# Why are some volcano plots flat?
+
+# In[8]:
+
+
+# Lets look at the expression data for the volcano plot with DEGs
+simulated_expression_filename = os.path.join(
+    local_dir,
+    "pseudo_experiment",
+    f"selected_simulated_data_{project_id}_11.txt"
+)
+
+simulated_expression = pd.read_csv(
+    simulated_expression_filename, 
+    sep="\t", 
+    header=0, 
+    index_col=0
+)
+
+simulated_expression
+
+
+# In[9]:
+
+
+sns.distplot(simulated_expression.loc["GSM822708_wtLB_A.CEL"])
+sns.distplot(simulated_expression.loc["GSM822712_delta_cbrBLB_A.CEL"])
+
+
+# In[10]:
+
+
+# Heatmap visualization
+sns.heatmap(simulated_expression.iloc[:,10:20])
```

Comment:
How are you choosing the genes to look at here? Are they just random ones?

Reply:
These were just randomly selected to get a little sense for the expression changes